### PR TITLE
Fix monitoring module for e2e tests

### PR DESCRIPTION
**Pull request by Rebits**
# Description

This PR include a fix for the monitoring module in E2E tests

## Testing performed

<details>

<summary> Testing </summary>

```
In [36]: monitoring_data =         {
In [36]: monitoring_data =         {
    ...:            "manager1":[
    ...:               {
    ...:                  "regex": r'(\d{4}\/\d{2}\/\d{2} \d{2}:\d{2}:\d{2}) .*? INFO: Testi
    ...: ng2',
    ...:                  "file":"/tmp/testing",
    ...:                  "timeout":10,
    ...:                  "n_iterations":1,
    ...:                  "greater_than_timestamp":"2024-02-20T17:12:22.33"
    ...:               }
    ...:            ]
    ...:         }

In [37]: monitoring_events_multihost(hm, monitoring_data)
Out[37]: 
{'manager1': {'not_found': [{'regex': '(\\d{4}\\/\\d{2}\\/\\d{2} \\d{2}:\\d{2}:\\d{2}) .*? INFO: Testing2',
    'file': '/tmp/testing',
    'timeout': 10,
    'n_iterations': 1,
    'greater_than_timestamp': '2024-02-20T17:12:22.33'}],
  'found': []}}

In [38]: monitoring_data =         {
    ...:            "manager1":[
    ...:               {
    ...:                  "regex": r'(\d{4}\/\d{2}\/\d{2} \d{2}:\d{2}:\d{2}) .*? INFO: Testi
    ...: ng',
    ...:                  "file":"/tmp/testing",
    ...:                  "timeout":10,
    ...:                  "n_iterations":1,
    ...:                  "greater_than_timestamp":"2024-02-20T17:12:22.33"
    ...:               }
    ...:            ]
    ...:         }

In [39]: monitoring_events_multihost(hm, monitoring_data)
Out[39]: {'manager1': {'not_found': [], 'found': ['2024/02/21 10:33:37']}}

In [40]: monitoring_data =         {
    ...:            "manager1":[
    ...:               {
    ...:                  "regex": r'Testing',
    ...:                  "file":"/tmp/testing",
    ...:                  "timeout":10,
    ...:                  "n_iterations":1,
    ...:                  "greater_than_timestamp":"2024-02-20T17:12:22.33"
    ...:               }
    ...:            ]
    ...:         }

In [41]: 

In [41]: monitoring_events_multihost(hm, monitoring_data)
---------------------------------------------------------------------------
ValueError           

In [42]: monitoring_data =         {
    ...:            "manager1":[
    ...:               {
    ...:                  "regex": r'Testing',
    ...:                  "file":"/tmp/testing",
    ...:                  "timeout":10,
    ...:                  "n_iterations":1,
    ...: 
    ...:               }
    ...:            ]
    ...:         }

In [43]: monitoring_events_multihost(hm, monitoring_data)
Out[43]: {'manager1': {'not_found': [], 'found': ['Testing', 'Testing', 'Testing']}}


In [44]: monitoring_data =         {
    ...:            "manager1":[
    ...:               {
    ...:                  "regex": r'(\d{4}\/\d{2}\/\d{2} \d{2}:\d{2}:\d{2}) .*? INFO: Testi
    ...: ng',
    ...:                  "file":"/tmp/testing",
    ...:                  "timeout":10,
    ...:                  "n_iterations":1,
    ...:                  "greater_than_timestamp":"2024-03-20T17:12:22.33"
    ...:               }
    ...:            ]
    ...:         }

In [45]: monitoring_events_multihost(hm, monitoring_data)
Out[45]: 
{'manager1': {'not_found': [{'regex': '(\\d{4}\\/\\d{2}\\/\\d{2} \\d{2}:\\d{2}:\\d{2}) .*? INFO: Testing',
    'file': '/tmp/testing',
    'timeout': 10,
    'n_iterations': 1,
    'greater_than_timestamp': '2024-03-20T17:12:22.33'}],
  'found': []}}


             
```

</details>

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py [hunk 1]
```diff
@@ -26,7 +26,8 @@
 from wazuh_testing.tools.system import HostManager
 
 
-def monitoring_events_multihost(host_manager: HostManager, monitoring_data: Dict, ignore_error: bool = False) -> Dict:
+def monitoring_events_multihost(host_manager: HostManager, monitoring_data: Dict, ignore_error: bool = True,
+                                scan_interval: int = 5) -> Dict:
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:30]: Should we ignore errors by default?

Comment (juliamagan) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:30]: `scan_interval` could use `T_5`. In addition, it is missing in the function documentation.

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py [hunk 2]
```diff
@@ -61,16 +62,17 @@ def monitoring_events_multihost(host_manager: HostManager, monitoring_data: Dict
            }
         }
     """
-    def monitoring_event(host_manager: HostManager, host: str, monitoring_elements: List[Dict], scan_interval: int = 20,
-                         ignore_error: bool = False) -> Dict:
+    def monitoring_event(host_manager: HostManager, host: str, monitoring_elements: List[Dict],
+                         ignore_error: bool = True,
+                         scan_interval: int = 5) -> Dict:
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:67]: Same here

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py [hunk 3]
```diff
@@ -95,10 +97,15 @@ def filter_events_by_timestamp(match_events: List) -> List:
...
 
-                timestamp_datetime = datetime.strptime(timestamp_str, timestamp_format)
-                greater_than_timestamp_formatted = datetime.strptime(greater_than_timestamp, timestamp_format_parameter)
+                try:
+                    timestamp_datetime = datetime.strptime(timestamp_str, timestamp_format)
+                    greater_than_timestamp_formatted = datetime.strptime(greater_than_timestamp,
+                                                                         timestamp_format_parameter)
+                except ValueError:
+                    raise ValueError(f"Timestamp format not supported: {timestamp_str}."
+                                     'Do the regex includes the timestamp?')
```
Comment (juliamagan) [inline deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:106]: If you already have the `greater_than_timestamp` field, is it necessary to add the timestamp regex to the `regex` field? Isn't it a bit repetitive?

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ!

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py (hunk 3 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:106]: To effectively compare with the time specified in the `greater_than_timestamp` parameters, we need to gather the timestamp from the event. Therefore, the specified regex should be designed to match at least one group, specifically tailored for capturing the timestamp.

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py (hunk 1 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:30]: **Regarding the ignore_error**

The current naming suggests the wrong functionality. It doesn't actually ignore errors; rather, when enabled, it prevents a TimeoutError from being triggered if some regex patterns aren't found. This functionality is particularly useful in the multithreading environment of E2E tests. Sometimes, we need to monitor various regex patterns on different hosts, and halting monitoring with an exception would hinder this process.
I suggest renaming this variable



**Regarding the use of `T_5`**

The use of the `T_5` is not recommended in this case due to:
- The import of IT modules could lead into unexpected errors. Check https://github.com/wazuh/wazuh-jenkins/issues/6234, for more information
- `T_5` is not a meaningful variable. I suggest using a variable like `DEFAULT_SCAN_INTERVAL`

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py (hunk 2 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:67]: Relatted conversation: https://github.com/wazuh/wazuh-qa/pull/4959#discussion_r1499090537

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py (hunk 3 above)
Comment (juliamagan) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py:106]: We will revisit this in the future

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py (hunk 1 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py]: Done d0068357a52f1d59580f3777ab4e35db30321570

File: deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py (hunk 2 above)
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/end_to_end/monitoring.py]: Done d0068357a52f1d59580f3777ab4e35db30321570

**Review by juliamagan — DISMISSED** [review]
LGTM

**Review by juliamagan — APPROVED** [review]

**Pull request merged**